### DISCUSSION TITLE
Is there a way to make provider2 dependent on provider1 using a class-based syntax for riverpod_generator?

### DISCUSSION THREAD

**Discussion by ToddManUtah** [rrousselGit/riverpod#2698] [Q&A]
I have 2 providers that have a dependency that works.

FiltersProvider

    part 'filters_provider.g.dart';
    
    enum Filter {
      glutenFree,
      lactoseFree,
      vegetarian,
      vegan,
    }
    
    @riverpod
    class Filters extends _$Filters {
      @override
      Map<Filter, bool> build() {
        return {
          Filter.glutenFree: false,
          Filter.lactoseFree: false,
          Filter.vegetarian: false,
          Filter.vegan: false,
        };
      }
    
      void updateFilter(Filter filter, bool value) {
        state = {...state}..update(filter, (_) => value);
      }
    }

MealsProvider

    part 'meals_provider.g.dart';
    
    @riverpod
    List<Meal> meals(MealsRef ref) {
      final filter = ref.watch(filtersProvider);
      final filteredMeals = dummyMeals.where((category) {
        if (filter[Filter.glutenFree]! && !category.isGlutenFree) {
          return false;
        }
        if (filter[Filter.lactoseFree]! && !category.isLactoseFree) {
          return false;
        }
        if (filter[Filter.vegetarian]! && !category.isVegetarian) {
          return false;
        }
        if (filter[Filter.vegan]! && !category.isVegan) {
          return false;
        }
        return true;
      }).toList();
      return filteredMeals;
    }

This works. The MealsProvider is updated automatically when the FiltersProvider changes.

Here is my problem: I want to convert MealsProvider to use a class based syntax for my code generation and I can't figure it out.

If I convert MealsProvider to a class based syntax, how do I watch the FiltersProvider. Here is my problem in code.

    part 'meals_provider.g.dart';
    
    @riverpod
    class Meals extends _$Meals {
      @override
      List<Meal> build() {
        return dummyMeals;
      }
    }

I'm not sure where to add the watch for the FiltersProvider. I can't put it into the "build()" method because I don't have access to a ref object. I like the flexibility of adding new helper methods to MealsProvider and so I want to use the class-based syntax but I'm just not sure where I can add the code to watch the FiltersProvider AND affect state in the MealsProvider.

Help!!!

**Comment by rrousselGit** [comment]
You can use `ref.watch` inside `Meals.build`

**Comment by ToddManUtah** [comment]
I KNOW I tried that and it didn't work. I just added this code and it worked great. THANKS!!!

    @riverpod
    class Meals extends _$Meals {
      @override
      List<Meal> build() {
        final filters = ref.watch(filtersProvider);
        return dummyMeals.where((meal) {
          if(filters[Filter.glutenFree]! && !meal.isGlutenFree) {
            return false;
          }
          if(filters[Filter.lactoseFree]! && !meal.isLactoseFree) {
            return false;
          }
          if(filters[Filter.vegetarian]! && !meal.isVegetarian) {
            return false;
          }
          if(filters[Filter.vegan]! && !meal.isVegan) {
            return false;
          }
          return true;
        }).toList();
      }
    }
